capstone-frontend/src/App.jsx

```diff
@@ -22,27 +21,45 @@ class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      searchQuery: ""
+      searchQuery: "",
+      bookLists: []
     };
   }
 
   setSearchQuery = (value) => {
     if (value !== this.state.searchQuery) {
-      this.setState({ searchQuery:value });
+      this.setState({ searchQuery: value });
     }
   }
 
+  fetchBookLists = async () => {
+
+    const userID = window.localStorage.getItem("userID");
+
+    const bookLists = await fetch(`/api/booklist?userID=${userID}`, {
+      method: "GET",
+    }).then(resp => resp.json()).catch(err => console.log(err));
+
+    this.setState({ bookLists });
+  }
+
+  componentDidMount() {
+    this.fetchBookLists();
+  }
+
+
   render() {
     return (
       <Router>
         <Navbar setSearchQuery={this.setSearchQuery} />
         <div className="row">
-          <LeftSideBar />
+          <LeftSideBar bookLists={this.state.bookLists} updateBookLists={this.fetchBookLists} />
           <div className="col-12 col-md-8" id="body-row">
             <Route exact path='/' component={Home} />
-            <Route path='/browse/:query' component={Browse} />
+            <Route path='/browse/:query' component={(props) => (
```

Comment:
Use the [`render` method](https://reactrouter.com/web/api/Route/render-func) instead of `component` here. This is likely the reason your component is unmounting then mounting again, giving you that error about setting state on an unmounted component. See the second paragraph [here](https://reactrouter.com/web/api/Route/component).

Reply:
ok